08-interop/08-interop.md

```diff
@@ -169,3 +169,123 @@ Task #10 - UDP between lwIP and emb6 on iotlab-m3
 Link-local UDP over IPv6 packet exchange (payload length 8) between an iotlab-m3
 node running RIOT with lwIP and an iotlab-m3 node running RIOT with emb6 (in
 both directions).
+
+Task #11 - UDP exchange between iotlab-m3 and Zephyr
+=====================================================
+### Description
+
+UDP packet exchange between an iotlab-m3 node running RIOT and a samr21-xpro
+node running Zephyr.
+
+### Testing procedure
+
+#### Setting up the Zephyr node
+
+1. Follow the
+['Getting Started'](https://docs.zephyrproject.org/latest/getting_started/index.html)
+guide on Zephyr's documentation.
+2. You will be using the `echo_server` example application. To compile and
+   flash, connect the samr21-xpro board and in zephyr's root directory run:
+
+   ```sh
+   west build -p auto -b atsamr21_xpro samples/net/sockets/echo_server -- -DOVERLAY_CONFIG=overlay-802154.conf
+   west flash
+   ```
+3. If everything is OK you should be able to connect to the serial output use
+   any terminal program. For example:
+   ```sh
+   minicom -D /dev/ttyACM0 -o
+   ```
+4. By pressing tab you should be able to see all available commands. Get the
+   node's IPv6 by running the following on its shell:
+
+   ```sh
+   # On samr21-xpro:zephyr node
+   uart:~$ net ipv6
+
+   IPv6 support                              : enabled
+   IPv6 fragmentation support                : disabled
+   Multicast Listener Discovery support      : enabled
+   Neighbor cache support                    : enabled
+   Neighbor discovery support                : enabled
+   Duplicate address detection (DAD) support : enabled
+   Router advertisement RDNSS option support : enabled
+   6lo header compression support            : enabled
+   Max number of IPv6 network interfaces in the system          : 1
+   Max number of unicast IPv6 addresses per network interface   : 3
+   Max number of multicast IPv6 addresses per network interface : 4
+   Max number of IPv6 prefixes per network interface            : 2
+
+   IPv6 addresses for interface 0x20007d60 (IEEE 802.15.4)
+   =======================================================
+   Type            State           Lifetime (sec)  Address
+   autoconf        preferred       infinite        fe80::d419:100:7ae4:9b3b/128
+   manual          preferred       infinite        2001:db8::1/128
+   ```
+
+   Similarly, to get the IEEE 802.15.4 PAN ID and channel, run:
+
+   ```sh
+   # On samr21-xpro:zephyr node
+   uart:~$ ieee802154 get_pan_id
+   PAN ID 43981 (0xabcd)
+
+   uart:~$ ieee802154 get_chan
+   Channel 26
+   ```
+
+#### Setting up the RIOT node
+
+1. Flash the [gnrc_networking] example or the [gnrc_udp] test to the iotlab-m3 board
+   test.
+2. Set the channel and PAN ID to the same values as the Zephyr node:
+   ```sh
+   # On iotlab-m3:riot node
+   > ifconfig 6 set chan 26
```

Comment:
With *gnrc_networking* the interface number is 7.

Reply:
This should be solved once https://github.com/RIOT-OS/RIOT/pull/12738 and we adapt the `ifconfig` shell command for it (which due to the test scripts might be a quite massive change). :sweat_smile: